src/GafferImage/RankFilter.cpp

```diff
@@ -371,13 +371,15 @@ IECore::ConstFloatVectorDataPtr RankFilter::computeChannelData( const std::strin
 	vector<float> &result = resultData->writable();
 	result.reserve( ImagePlug::tileSize() * ImagePlug::tileSize() );
 
+	// TODO - while thinking about lifetimes ... doesn't holding a reference to the getValue() return
+	// have a tiny chance of crashing if the string is evicted from the cache?
```

Comment:
No, the string isn't in the cache, it's a temporary copy returned by `StringPlug::getValue()`. And the C++ lifetime rules extend the lifetime of the temporary to match the lifetime of the reference : https://en.cppreference.com/w/cpp/language/reference_initialization#Lifetime_of_a_temporary. I hate this "assign temporary to const &" pattern though - it's confusing, and I _think_ the compiler can just optimise out any copy if you do the less confusing `const string = `.

Reply:
Removed comment in e15eea3.  You didn't seem to confident in whether switching to `const string =` was the correct approach, so I've left it as-is.